2006-10-21  Guillem Jover  <guillem@hadrons.org>

	* libinetutils/localhost.c (localhost): Determine the FQDN only
	if gethostname did not fail.


Index: b/libinetutils/localhost.c
===================================================================
--- a/libinetutils/localhost.c
+++ b/libinetutils/localhost.c
@@ -77,22 +77,23 @@ localhost (void)
       free (buf);
       buf = 0;
     }
+  else
+    /* Determine FQDN */
+    {
+      struct hostent *hp = gethostbyname(buf);
 
-  /* Determine FQDN */
-  {
-    struct hostent *hp = gethostbyname(buf);
+      if (hp)
+        {
+	  struct in_addr addr;
+	  addr.s_addr = *(unsigned int*) hp->h_addr;
+	  hp = gethostbyaddr((char *) &addr, sizeof(addr), AF_INET);
+	  if (hp)
+	    {
+	      free(buf);
+	      buf = strdup(hp->h_name);
+	    }
+        }
+    }
 
-    if (hp)
-      {
-	struct in_addr addr;
-	addr.s_addr = *(unsigned int*) hp->h_addr;
-	hp = gethostbyaddr((char *) &addr, sizeof(addr), AF_INET);
-	if (hp)
-	  {
-	    free(buf);
-	    buf = strdup(hp->h_name);
-	  }
-      }
-  }
   return buf;
 }
